Sources/HarmonizeSemantics/GrammarComponents/Decoration/Assignment.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  Assignment.swift
+//  Harmonize
+//
+//  Copyright 2024 Perry Street Software Inc.
+
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+
+//  http://www.apache.org/licenses/LICENSE-2.0
+
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import SwiftSyntax
+
+/// Represents an assignment expression in Swift syntax, specifically an infix operator assignment (e.g., `a = b`).
+public struct Assignment: DeclarationDecoration, SyntaxNodeProviding {
+    /// The underlying syntax node for the assignment expression.
+    public let node: InfixOperatorExprSyntax
+    
+    /// The target of the assignment expression (the left operand).
+    public var target: String {
+        node.leftOperand.trimmedDescription
+    }
+    
+    /// the assignment value after `=` (the right operand)
+    public var rightOperand: RightOperand {
```

Comment:
Would it make sense to call them either `target` & `value` or `leftOperand` & `rightOperand`?